src/types.ts

```diff
@@ -0,0 +1,399 @@
+import Constants from './constants';
+import { parseNumber, valueof } from './utils';
+
+interface IdentitiesByType {
+    [key: number]: string;
+}
+
+export const MessageType = {
+    SessionStart: 1 as const,
+    SessionEnd: 2 as const,
+    PageView: 3 as const,
+    PageEvent: 4 as const,
+    CrashReport: 5 as const,
+    OptOut: 6 as const,
+    AppStateTransition: 10 as const,
+    Profile: 14 as const,
+    Commerce: 16 as const,
+    Media: 20 as const,
+    UserAttributeChange: 17 as const,
+    UserIdentityChange: 18 as const,
+};
+
+// Dictionary that contains MessageTypes that will
+// trigger an immediate upload.
+export const TriggerUploadType = {
+    [MessageType.Commerce]: 1 as const,
```

Comment:
if these are types, should they have the same value `1`?

Reply:
This is something that probably needs to be refactored.  Were [using the values as a sort of lookup table](https://github.com/mParticle/mparticle-web-sdk/blob/master/src/batchUploader.ts#L186) in the batchUploader when it might be cleaner to do this as a function instead. But that is likely out of scope for this PR.